### ISSUE TITLE
elf symtab_entries

### ISSUE THREAD

**Issue by lprat** [VirusTotal/yara#1394]
Hi,
I tried to use "symtab_entries" on elf, but not work on version 4.0.2.
Can you help me to find why my rule dont work?

My rules:
```
import "elf"
rule test1 { 
        condition: 
          elf.symtab_entries > 0
}
rule test2 { 
        condition: 
        for any i in (0..elf.symtab_entries):
        (
            elf.symtab[i].bind == elf.STB_GLOBAL
        )
}
```
Tried to use but dont work:
```
$ yara -v
4.0.2
$ readelf -Ws /bin/ls

La table de symboles « .dynsym » contient 128 entrées :
   Num:    Valeur         Tail Type    Lien   Vis      Ndx Nom
     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND 
     1: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND __ctype_toupper_loc@GLIBC_2.3 (2)
     2: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND getenv@GLIBC_2.2.5 (3)
     3: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fgetfilecon@LIBSELINUX_1.0 (4)
     4: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND sigprocmask@GLIBC_2.2.5 (3)
     5: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND __snprintf_chk@GLIBC_2.3.4 (5)
     6: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND raise@GLIBC_2.2.5 (3)
     7: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND abort@GLIBC_2.2.5 (3)
$yara rules.yar /bin/ls
$yara -D ../rules.yar /bin/ls
elf
	symtab
	symtab_entries = YR_UNDEFINED
	dynamic
		[0]
			type = 1
			val = 1359
		[1]
			type = 1
			val = 1375
		[2]
			type = 12
			val = 16384
		[3]
			type = 13
			val = 98720
		[4]
			type = 25
			val = 144208
		[5]
			type = 27
			val = 8
		[6]
			type = 26
			val = 144216
		[7]
			type = 28
			val = 8
		[8]
			type = 1879047925
			val = 776
		[9]
			type = 5
			val = 4024
		[10]
			type = 6
			val = 952
		[11]
			type = 10
			val = 1477
		[12]
			type = 11
			val = 24
		[13]
			type = 21
			val = 0
		[14]
			type = 3
			val = 147456
		[15]
			type = 2
			val = 2520
		[16]
			type = 20
			val = 7
		[17]
			type = 23
			val = 11104
		[18]
			type = 7
			val = 5920
		[19]
			type = 8
			val = 5184
		[20]
			type = 9
			val = 24
		[21]
			type = 1879048187
			val = 134217728
		[22]
			type = 1879048190
			val = 5760
		[23]
			type = 1879048191
			val = 2
		[24]
			type = 1879048176
			val = 5502
		[25]
			type = 1879048185
			val = 203
		[26]
			type = 0
			val = 0
	dynamic_section_entries = 27
	segments
		[0]
			type = 6
			flags = 4
			offset = 64
			virtual_address = 64
			physical_address = 64
			file_size = 616
			memory_size = 616
			alignment = 8
		[1]
			type = 3
			flags = 4
			offset = 680
			virtual_address = 680
			physical_address = 680
			file_size = 28
			memory_size = 28
			alignment = 1
		[2]
			type = 1
			flags = 4
			offset = 0
			virtual_address = 0
			physical_address = 0
			file_size = 13624
			memory_size = 13624
			alignment = 4096
		[3]
			type = 1
			flags = 5
			offset = 16384
			virtual_address = 16384
			physical_address = 16384
			file_size = 82345
			memory_size = 82345
			alignment = 4096
		[4]
			type = 1
			flags = 4
			offset = 102400
			virtual_address = 102400
			physical_address = 102400
			file_size = 36616
			memory_size = 36616
			alignment = 4096
		[5]
			type = 1
			flags = 6
			offset = 140112
			virtual_address = 144208
			physical_address = 144208
			file_size = 4728
			memory_size = 9576
			alignment = 4096
		[6]
			type = 2
			flags = 6
			offset = 142808
			virtual_address = 146904
			physical_address = 146904
			file_size = 496
			memory_size = 496
			alignment = 8
		[7]
			type = 4
			flags = 4
			offset = 708
			virtual_address = 708
			physical_address = 708
			file_size = 68
			memory_size = 68
			alignment = 4
		[8]
			type = 1685382480
			flags = 4
			offset = 123684
			virtual_address = 123684
			physical_address = 123684
			file_size = 2372
			memory_size = 2372
			alignment = 4
		[9]
			type = 1685382481
			flags = 6
			offset = 0
			virtual_address = 0
			physical_address = 0
			file_size = 0
			memory_size = 0
			alignment = 16
		[10]
			type = 1685382482
			flags = 4
			offset = 140112
			virtual_address = 144208
			physical_address = 144208
			file_size = 3248
			memory_size = 3248
			alignment = 1
	PF_R = 4
	PF_W = 2
	PF_X = 1
	STB_WEAK = 2
	STB_GLOBAL = 1
	STB_LOCAL = 0
	STT_TLS = 6
	STT_COMMON = 5
	STT_FILE = 4
	STT_SECTION = 3
	STT_FUNC = 2
	STT_OBJECT = 1
	STT_NOTYPE = 0
	DT_ENCODING = 32
	DT_FLAGS = 30
	DT_RUNPATH = 29
	DT_FINI_ARRAYSZ = 28
	DT_INIT_ARRAYSZ = 27
	DT_FINI_ARRAY = 26
	DT_INIT_ARRAY = 25
	DT_BIND_NOW = 24
	DT_JMPREL = 23
	DT_TEXTREL = 22
	DT_DEBUG = 21
	DT_PLTREL = 20
	DT_RELENT = 19
	DT_RELSZ = 18
	DT_REL = 17
	DT_SYMBOLIC = 16
	DT_RPATH = 15
	DT_SONAME = 14
	DT_FINI = 13
	DT_INIT = 12
	DT_SYMENT = 11
	DT_STRSZ = 10
	DT_RELAENT = 9
	DT_RELASZ = 8
	DT_RELA = 7
	DT_SYMTAB = 6
	DT_STRTAB = 5
	DT_HASH = 4
	DT_PLTGOT = 3
	DT_PLTRELSZ = 2
	DT_NEEDED = 1
	DT_NULL = 0
	PT_GNU_STACK = 1685382481
	PT_GNU_EH_FRAME = 1685382480
	PT_TLS = 7
	PT_PHDR = 6
	PT_SHLIB = 5
	PT_NOTE = 4
	PT_INTERP = 3
	PT_DYNAMIC = 2
	PT_LOAD = 1
	PT_NULL = 0
	sections
		[0]
			type = 0
			flags = 0
			address = 0
			name = ""
			size = 0
			offset = 0
		[1]
			type = 1
			flags = 2
			address = 680
			name = ".interp"
			size = 28
			offset = 680
		[2]
			type = 7
			flags = 2
			address = 708
			name = ".note.gnu.build-id"
			size = 36
			offset = 708
		[3]
			type = 7
			flags = 2
			address = 744
			name = ".note.ABI-tag"
			size = 32
			offset = 744
		[4]
			type = 1879048182
			flags = 2
			address = 776
			name = ".gnu.hash"
			size = 172
			offset = 776
		[5]
			type = 11
			flags = 2
			address = 952
			name = ".dynsym"
			size = 3072
			offset = 952
		[6]
			type = 3
			flags = 2
			address = 4024
			name = ".dynstr"
			size = 1477
			offset = 4024
		[7]
			type = 1879048191
			flags = 2
			address = 5502
			name = ".gnu.version"
			size = 256
			offset = 5502
		[8]
			type = 1879048190
			flags = 2
			address = 5760
			name = ".gnu.version_r"
			size = 160
			offset = 5760
		[9]
			type = 4
			flags = 2
			address = 5920
			name = ".rela.dyn"
			size = 5184
			offset = 5920
		[10]
			type = 4
			flags = 66
			address = 11104
			name = ".rela.plt"
			size = 2520
			offset = 11104
		[11]
			type = 1
			flags = 6
			address = 16384
			name = ".init"
			size = 23
			offset = 16384
		[12]
			type = 1
			flags = 6
			address = 16416
			name = ".plt"
			size = 1696
			offset = 16416
		[13]
			type = 1
			flags = 6
			address = 18112
			name = ".plt.got"
			size = 24
			offset = 18112
		[14]
			type = 1
			flags = 6
			address = 18144
			name = ".text"
			size = 80574
			offset = 18144
		[15]
			type = 1
			flags = 6
			address = 98720
			name = ".fini"
			size = 9
			offset = 98720
		[16]
			type = 1
			flags = 2
			address = 102400
			name = ".rodata"
			size = 21281
			offset = 102400
		[17]
			type = 1
			flags = 2
			address = 123684
			name = ".eh_frame_hdr"
			size = 2372
			offset = 123684
		[18]
			type = 1
			flags = 2
			address = 126056
			name = ".eh_frame"
			size = 12960
			offset = 126056
		[19]
			type = 14
			flags = 3
			address = 144208
			name = ".init_array"
			size = 8
			offset = 140112
		[20]
			type = 15
			flags = 3
			address = 144216
			name = ".fini_array"
			size = 8
			offset = 140120
		[21]
			type = 1
			flags = 3
			address = 144224
			name = ".data.rel.ro"
			size = 2680
			offset = 140128
		[22]
			type = 6
			flags = 3
			address = 146904
			name = ".dynamic"
			size = 496
			offset = 142808
		[23]
			type = 1
			flags = 3
			address = 147400
			name = ".got"
			size = 56
			offset = 143304
		[24]
			type = 1
			flags = 3
			address = 147456
			name = ".got.plt"
			size = 864
			offset = 143360
		[25]
			type = 1
			flags = 3
			address = 148320
			name = ".data"
			size = 616
			offset = 144224
		[26]
			type = 8
			flags = 3
			address = 148960
			name = ".bss"
			size = 4824
			offset = 144840
		[27]
			type = 1
			flags = 0
			address = 0
			name = ".gnu_debugaltlink"
			size = 73
			offset = 144840
		[28]
			type = 1
			flags = 0
			address = 0
			name = ".gnu_debuglink"
			size = 52
			offset = 144916
		[29]
			type = 3
			flags = 0
			address = 0
			name = ".shstrtab"
			size = 284
			offset = 144968
	ph_entry_size = 56
	ph_offset = 64
	number_of_segments = 11
	sh_entry_size = 64
	sh_offset = 145256
	number_of_sections = 30
	entry_point = 24928
	machine = 62
	type = 3
	SHF_EXECINSTR = 4
	SHF_ALLOC = 2
	SHF_WRITE = 1
	SHT_DYNSYM = 11
	SHT_SHLIB = 10
	SHT_REL = 9
	SHT_NOBITS = 8
	SHT_NOTE = 7
	SHT_DYNAMIC = 6
	SHT_HASH = 5
	SHT_RELA = 4
	SHT_STRTAB = 3
	SHT_SYMTAB = 2
	SHT_PROGBITS = 1
	SHT_NULL = 0
	EM_AARCH64 = 183
	EM_X86_64 = 62
	EM_ARM = 40
	EM_PPC64 = 21
	EM_PPC = 20
	EM_MIPS_RS3_LE = 10
	EM_MIPS = 8
	EM_860 = 7
	EM_88K = 5
	EM_68K = 4
	EM_386 = 3
	EM_SPARC = 2
	EM_M32 = 1
	EM_NONE = 0
	ET_CORE = 4
	ET_DYN = 3
	ET_EXEC = 2
	ET_REL = 1
	ET_NONE = 0

...
Where is my error?

Thanks for your help.

**Comment by lprat** [comment]
Ok i understand the problem, You dont use symtab on 64bits binary, just 32bits:

```
libocijdbc19.so: ELF 32-bit LSB shared object, Intel 80386, version 1 (SYSV), dynamically linked, BuildID[sha1]=bbf7216af5910a35da566de190df42f141a30e3a, not stripped

ls: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=6461a544c35b9dc1d172d1a1c09043e487326966, for GNU/Linux 3.2.0, stripped

$yara rules.yar /bin/ls
$
$yara rules.yar libocijdbc19.so 
test1 libocijdbc19.so
test2 libocijdbc19.so
```

Is it normal work (not use on x64)?

Thanks

**Comment by plusvic** [comment]
Implemented in #1427.
